### PR TITLE
Dev favcreation vuex

### PR DESCRIPTION
**Pull request by AnssKF**

### CODE REVIEW

File: src/nova-components/FavCreation/resources/js/store/index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import FavStore from './modules/favorite-store';
+import UserStore from './modules/user-store';
+
+export function registerStoreModules(store){
+    store.registerModule('FavStore', FavStore)
```
Comment (Mhany95) [inline src/nova-components/FavCreation/resources/js/store/index.js:5]: follow naming convention and other good practices
https://vuejs.org/v2/style-guide/

File: src/nova-components/ComponentUi/resources/js/components/Tool.vue [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<template>
+  <div>Component Ui</div>
+</template>
+
+<script>
+export default {
+  props: ['resourceName', 'resourceId', 'panel'],
```
Comment (AbdElshorafa) [inline src/nova-components/ComponentUi/resources/js/components/Tool.vue:7]: What's the usage of this props?

File: src/nova-components/ComponentUi/resources/js/components/ui/c-btn/CBtn.vue [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<template>
+    <button 
+        @click="handleOnClick"
+        v-bind="$attrs"
+        class="c-btn c-btn-shadow-effect"
+        :class="getBtnStyle">
```
Comment (AbdElshorafa) [inline src/nova-components/ComponentUi/resources/js/components/ui/c-btn/CBtn.vue:6]: Combine the two classes in one class property

File: src/nova-components/ComponentUi/routes/api.php [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+|
+| Here is where you may register API routes for your tool. These routes
+| are loaded by the ServiceProvider of your tool. You're free to add
+| as many additional routes to this file as your tool may require.
+|
+*/
+
+// Route::get('/endpoint', function (Request $request) {
+//     //
+// });
```
Comment (AbdElshorafa) [inline src/nova-components/ComponentUi/routes/api.php:19]: Remove comments

File: src/nova-components/FavCreation/resources/js/store/modules/user-store/actions.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                search: searchResult,
+                first: false,
+                withTrashed: false,
+                viaResource: '',
+                viaResourceId: '',
+            },
+            cancelToken: searchCancelTokenSource.token
+        })
+
+        commit(SET_AVAILABLE_USERS, res.data.resources)
```
Comment (AbdElshorafa) [inline src/nova-components/FavCreation/resources/js/store/modules/user-store/actions.js:32]: Add guards to {res, res.data, res.data.resource }

File: src/nova-components/FavCreation/resources/js/store/modules/favorite-store/actions.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { UPDATE_FAV_FORM_FIELD, RESET_FAV_FORM } from './mutations'
+
+const addFav = async ({ commit, state }) => {
+    const url           = state.favForm.user.value;
```
Comment (Mhany95) [inline src/nova-components/FavCreation/resources/js/store/modules/favorite-store/actions.js:4]: object destructing

File: src/nova-components/FavCreation/resources/js/store/modules/user-store/actions.js [hunk 7]
```diff
@@ -24,22 +25,20 @@ const fetchUsers = async ({ commit, state }, searchResult) => {
...
             cancelToken: searchCancelTokenSource.token
         })
 
-        if(!res.data.resources) return Promise.reject()
+        if(!res.data.resources) return Promise.reject('Invalid Response');
         
-        commit(SET_AVAILABLE_USERS, res.data.resources)
+        const availableUsers = UsersHelper.parseUserResources(res.data.resources)
+        commit(SET_AVAILABLE_USERS, availableUsers)
+
```
Comment (AbdElshorafa) [inline src/nova-components/FavCreation/resources/js/store/modules/user-store/actions.js]: Guards for {res, res.data, res.data.resources}

**Pull request merged**